pkg/utils/openshift/webconsole_test.go

```diff
@@ -0,0 +1,82 @@
+package openshift
+
+import (
+	"github.com/ghodss/yaml"
+	oappsv1 "github.com/openshift/api/apps/v1"
+	v1 "github.com/openshift/api/console/v1"
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+func TestGetConsoleYAMLSample(t *testing.T) {
+	var inputYaml = `
+apiVersion: app.kiegroup.org/v2
+kind: KieApp
+metadata:
+ name: rhdm-trial
+ annotations:
+   consoleTargetAPIVersion: apps/v1
+   consoleTargetKind: Deployment
+   consoleName: kieapp-rhdm-trial
+   consoleDesc: KieApp rhdm trial
+   consoleTitle: KieApp rhdm trial
+spec:
+ environment: rhpam-trial
+`
+	resource := &oappsv1.DeploymentConfig{}
+	assert.NoError(t, yaml.Unmarshal([]byte(inputYaml), resource))
+
+	yamlSample, err := GetConsoleYAMLSample(resource)
+	assert.NoError(t, err)
+
+	assert.Equal(t, "kieapp-rhdm-trial", yamlSample.ObjectMeta.Name)
+	assert.Equal(t, "openshift-console", yamlSample.ObjectMeta.Namespace)
+	assert.Equal(t, "apps/v1", yamlSample.Spec.TargetResource.APIVersion)
+	assert.Equal(t, "Deployment", yamlSample.Spec.TargetResource.Kind)
+	assert.Equal(t, v1.ConsoleYAMLSampleTitle("KieApp rhdm trial"), yamlSample.Spec.Title)
+	assert.Equal(t, v1.ConsoleYAMLSampleDescription("KieApp rhdm trial"), yamlSample.Spec.Description)
+
+	yamlContent := yamlSample.Spec.YAML
+	res := &oappsv1.DeploymentConfig{}
+	assert.NoError(t, yaml.Unmarshal([]byte(string(yamlContent)), res))
+	assert.Empty(t, res.Annotations, "Annotations should be cleaned up.")
+	assert.Equal(t, "rhdm-trial", res.ObjectMeta.Name)
+	assert.Equal(t, "app.kiegroup.org/v2", res.APIVersion)
+	assert.Equal(t, "KieApp", res.Kind)
+}
+
+func TestGetConsoleYAMLSampleWithNoAnnotations(t *testing.T) {
+	var inputYaml = `
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: nginx-deployment
+  labels:
+    app: nginx
+spec:
+  replicas: 3
+`
+	resource := &oappsv1.DeploymentConfig{}
```

Comment:
Should be Deployment as per above, assuming the above is even correct (I have doubts about api version